docs/design-docs/06-config-as-data.md

```diff
@@ -0,0 +1,164 @@
+  serialized packge wire-format
+* uses a function `ResourceList → ResultList` (`kpt` function) as the
+  foundational, composable unit of package-manipulation code (note that other
+  forms of code can manipulate packages as well, i.e. UIs, custom algorithms
+  not necessarily packaged and used as kpt functions)
+
+and provides the following basic functionality:
+
+* load a serialized package from a repository (as `ResourceList`) (examples of
+  repository may be one or more of: local HDD, Git repository, OCI, Cloud
+  Storage, etc.)
+* save a serialized package (as `ResourceList`) to a package repository
+* evaluate a function on a serialized package (`ResourceList`)
+* [render](https://kpt.dev/book/04-using-functions/01-declarative-function-execution)
+  a package (evaluate functions declared within the package itself)
+* create a new (empty) package
+* fork (or clone) an existing package from one package repository (called
+  upstream) to another (called downstream)
+* delete a package from a repository
+* associate a version with the package; guarantee immutability of packages with
+  an assigned version
+* incorporate changes from the new version of an upstream package into a new
+  version of a downstream package
+* revert to a prior version of a package
+
+## Value
+
+The Config as Data approach enables some key value which is available in other
+configuration management approaches to a lesser extent or is not available
+at all.
+
+*CaD* approach enables:
+
+* simplified authoring of configuration using a variety of methods and sources
+* WYSIWYG interaction with configuration using a simple data serialization
+  formation rather than a code-like format
+* layering of interoperable interface surfaces (notably GUI) over declarative
+  configuration mechanisms rather than forcing choices between exclusive
+  alternatives (exclusively UI/CLI or IaC initially followed by exclusively
+  UI/CLI or exclusively IaC)
+* the ability to apply UX techniques to simplify configuration authoring and
+  viewing
+* compared to imperative tools (e.g., UI, CLI) that directly modify the live
+  state via APIs, CaD enables versioning, undo, audits of configuration history,
+  review/approval, pre-deployment preview, validation, safety checks,
+  constraint-based policy enforcement, and disaster recovery
+* bulk changes to configuration data in their sources of truth
+* injection of configuration to address horizontal concerns
+* merging of multiple sources of truth
+* state export to reusable blueprints without manual templatization
+* cooperative editing of configuration by humans and automation, such as for
+  security remediation (which is usually implemented against live-state APIs)
+* reusability of configuration transformation code across multiple bodies of
+  configuration data containing the same resource types
+* amortizing the effort of writing, testing, documenting the code
```

Comment:
This should be a sub-bullet, or incorporated into the bullet above

Reply:
fixed; thanks!